src/panels/lovelace/views/hui-view.ts

```diff
@@ -226,59 +226,64 @@ export class HUIView extends LitElement {
       root.removeChild(root.lastChild);
     }
 
-    let columns: [number, number][][] = [];
     const columnEntityCount: number[] = [];
+    const columnElements: HTMLDivElement[] = [];
     for (let i = 0; i < this.columns!; i++) {
-      columns.push([]);
+      const columnEl = document.createElement("div");
+      columnEl.classList.add("column");
+      root.appendChild(columnEl);
       columnEntityCount.push(0);
+      columnElements.push(columnEl);
     }
 
-    this._cards.forEach((el, index) => {
-      const cardSize = computeCardSize(
+    let cardsAdded = 0;
+
+    for (const [index, el] of this._cards.entries()) {
+      const cardSizeProm = computeCardSize(
         (el.tagName === "HUI-CARD-OPTIONS" ? el.firstChild : el) as LovelaceCard
```

Comment:
Would it be faster to check `this.lovelace!.editMode` here instead of `el.tagName` ? Even better if we store editMode as a local variable.

Reply:
I think it is not even needed anymore